tests/unit_tests/test_engine.py

```
@@ -367,3 +387,1790 @@ def test_add_health_check_not_callable(test_app: Engine):
+        assert result.total_notifiers == 2
+        assert result.notifiers_with_attachments == 1
+        assert result.notifiers_without_attachments == 1
+        assert result.notifiers_skipped == 0
+
+    def test_notification_result_with_skip_policy(self, test_app: Engine):
+        """Test NotificationResult with SKIP_NOTIFIER policy."""
+
+        def legacy_notifier(_message: str) -> None:
+            pass
+
+        def modern_notifier(_message: str, attachments: list[Attachment] | None = None) -> None:  # noqa: ARG001
```

Ruff (0.14.13)

Unused `noqa` directive (unused: `ARG001`)

Remove unused `noqa` directive

(RUF100)